src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -168,6 +168,8 @@ private ImmutableCollections() { }
     static <E> List<E> listCopy(Collection<? extends E> coll) {
         if (coll instanceof List12 || (coll instanceof ListN && ! ((ListN<?>)coll).allowNulls)) {
             return (List<E>)coll;
+        } else if (coll.isEmpty()) { // implicit nullcheck of coll
+            return List.of();
         } else {
             return (List<E>)List.of(coll.toArray()); // implicit nullcheck of coll
```

Comment:
The comment is no longer relevant here, as it now happens on line 171.

Reply:
@szegedi Nice catch, Attila. Corrected!

Reply:
Thanks @szegedi for catching this and @viktorklang-ora for fixing it. I like having comments like this in cases where we need to throw NPE for null and for which there's no explicit `Objects.requireNonNull`. We've had cases in the past where an apparently innocuous refactoring postponed an implicit nullcheck, which opened the possibility of a side effect occuring before NPE was thrown (violates failure idempotency). So I think maintaining such comments is important.

With that in mind, for the Set and Map cases, could you (Viktor) add similar comments there? Arguably they should have been there already, but, oh well, they weren't. Thanks.

Reply:
@stuart-marks Makes sense. I've added those comments to Set and Map copyOf()

Reply:
Great, thanks!